### PR TITLE
Totally restructured folders for the boilerplate

### PR DESCRIPTION
**Pull request by grafitto**
I have completely restructured the folders. I have also added a minimal documentation detailing the purpose of each folder.
Also added tests and linting support.

### CODE REVIEW

File: docs/FOLDERS.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+### api/common
+This folder contains all the logic that is shared across multiple files in the parent folder ([api](#api)).
+
+## config
+This is where configurations are loaded, configurations could be loaded from the environment variables either in production or in development based on where the service has been spawned.
+
+## docs
+All the docs for this service should be here.
+
+## Events
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: use small letter

Comment (grafitto) [reply docs/FOLDERS.md]: On it..

File: docs/FOLDERS.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+This is where all the events handlers should be located. Events are very ideal for queue related operations.
+
+## lib
+Contains the wrappers to external libraries.
+
+## scripts
+This contains all scripts that are run once or periodically, E.g. Migration scripts, indice creators etc.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: not periodically, just once
one time adminstrative scripts or local test utitlites as well any build configs

File: docs/FOLDERS.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## services
+This is where all the business code should be located.
+
+### services/models
+This should contain the schemas
+
+## workers
+All background logic E.g. cron jobs e.t.c should be located in this folder.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: Should contain anything that needs to run outside of the main process

File: docs/FOLDERS.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+These other files include:
+
+- #### .editorconfig
+
+    These are are editor configurations, just to make development smooth and standard across the board.
+
+- #### .env.sample
+
+    Has the samples for environment variable.   
+    A `.env` file __must__ be created based on the variables in the _.env.sample_ file.
```
Comment (obengwilliam) [inline docs/FOLDERS.md]: oonly in development

**Pull request merged**